streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -747,6 +748,18 @@ public void setUncaughtExceptionHandler(final Thread.UncaughtExceptionHandler eh
     }
 
     /**
+     * Set the listener which is triggered whenever a {@link StateStore} is being restored in order to resume
+     * processing.
+     *
+     * @param stateRestoreListener The listener triggered when {@link StateStore} is being restored.
+     */
+    public void setStateRestoreListener(final StateRestoreListener stateRestoreListener) {
```

Comment:
If we put it in config then they lose the ability to use capture any objects/state etc in their application.
We could always only allow the listener to be set when KafkaStreams is in the `CREATED` state and throw an exception if it isn't.